Documentation/project-docs/developer-guide.md

```diff
@@ -16,4 +11,235 @@ The CoreFX repo can be built from a regular, non-admin command prompt. The build
+```
+
+- The following example will **fail**, because the property `/p:skiptests=true` only applies to the build-managed component.
+```
+build -debug /p:skiptests=true     => ERROR!
+```
+
+###Build Native
+The native build produces shims over libc, openssl, gssapi, libcurl and libz.
+The build system uses CMake (2.8.12 or higher) to generate Makefiles using clang (3.5 or higher).
+The build also uses git for generating some version information.
+
+The native component should be buildable on any system.
+
+**Examples**
+
+- Building in debug mode for platform x64
+```
+build-native -debug -buildArch=x64
+```
+
+- The following example shows the argument `--`. Everything that is after it is not going to be processed and it is going to be passed as it is.
+```
+build-native -debug -buildArch=arm -- cross verbose
+```
+
+For more information about extra parameters take a look at the scripts `build-native` under src/Native.
+
+##Build Managed
+Since the managed build uses the .NET Core CLI (which the build will download), managed components can only be built on a subset of distros.
+There are some additional pre-requesties from the CLI which need to be installed. Both libicu and
+libunwind are used by CoreCLR to execute managed code, so they must be
+installed. Since CoreFX does not actually link against these packages, runtime
+versions are sufficent.  We also require curl to be present, which we use to
+download the .NET Core CLI.
+
+**Examples**
+
+- Building in debug mode for platform x64
+```
+build-managed -debug -buildArch=x64
+```
+
+- Building in debug mode for platform x64 targeting OS Linux
+```
+build-managed -debug -buildArch=x64 -os=Linux
+```
+
+- Building only the managed libraries with /flp:v=diag. It doesn't restore packages, builds tests or builds packages.
+```
+build-managed -binaries -verbose
+```
+
+- Building the managed libraries and the tests but don't run the tests.
+```
+build-managed -skiptests
+```
+
+- The following example shows the argument `--`. Everything that is after it is not going to be processed and it is going to be passed as it is.
+Use it to pass extra msbuild properties.
+```
+build-managed -binaries -- /p:WithoutCategories=IgnoreForCI
+```
+
+### Building individual CoreFx DLLs
+
+Under the src directory is a set of directories, each of which represents a particular assembly in CoreFX.
+  
+For example the src\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly. Each of these directories includes two projects, one for the DLL being built and one for the tests, both specified by a .builds file.
+
+You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the src\System.Diagnostics.DiagnosticsSource\src directory and typing `msbuild System.Diagnostics.DiagnosticSource.builds`. The DLL ends up as  bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource\System.DiagnosticSource.dll.
```

Comment:
We really need to provide some instructions for how to build individual projects on other OS's as well. Today I think the best we have is for people to directly call the msbuild scripts we have in our tools directory which should do the right things. Eventually we need to make a little more first class via our run commands.